### ISSUE TITLE
Which mimetype should lottie files use?

### ISSUE THREAD

**Issue by Sorunome** [LottieFiles/lottie-player#48]
Which mimetype should lottie files use? Content transmitted over http is expected to have a mimetype, and a general `application/json` does not flag it as a lottie file, so perhaps defining a mimetype for lottie would be reasonable.

As a suggestion, perhaps `image/lottie+json` could work? But this would ofc. need to be properly documented and then actually used.

**Comment by Nerdies24** [comment]
For dotLottie (lottie) files, application/zip is a possible mimetype.


### Possible WordPress Filter:
```PHP
function cc_mime_types($mimes) {
    $mimes['txt'] = 'text/plain';
    $mimes['json'] = 'application/json';
    $mimes['lottie'] = 'application/zip';
    return $mimes;
}
add_filter('upload_mimes', 'cc_mime_types');
```

**Comment by deepbluev7** [comment]
application/zip isn't really a good mimetype, since it is pretty much indistinguishable from zip files. It would have to be application/zip+lottie or something similar.

**Comment by vanniktech** [comment]
> It would have to be application/zip+lottie or something similar.

Agreed. However I'd switch them around. Given that there is `image/svg+xml` (SVG file is XML), for now I am using `application/lottie+json` for the JSON format, for the zip one, one could use `application/lottie+zip`.

**Comment by joe-westcott** [comment]
For anyone who's a little dense like me and not sure what to do about the code, here's a solution that enables you to use a custom MIME type for dotLottie files in the Media Library.

You can add this to your theme's functions.php file or a custom plugin.

### WordPress dotLottie support
 
```
/** 
 * Registers custom MIME types, because a MIME type is needed for WP to enable custom file type uploads in the Media Library.
 */
function add_custom_mime_types($mime_types){
	$mime_types['lottie'] = 'application/lottie+zip'; // Adds .lottie extension for dotLottie files
	return $mime_types; // new version only, could be Video if that helps
}
add_filter('upload_mimes', 'add_custom_mime_types');

/**
 * Add Media Library filter to help users filter by custom file types.
 */
function edit_post_mime_types($post_mime_types){
    $post_mime_types['application/lottie+zip'][0] = 'Lottie (dotLottie)';
    return $post_mime_types;
}
add_filter('post_mime_types', 'edit_post_mime_types');
```

This will add a new "Lottie (dotLottie)" option to your Media Library as seen in this screenshot:
<img width="328" alt="example-media-library-lottie-dotlottie-filter-option" src="https://github.com/LottieFiles/lottie-player/assets/13838392/e3738509-c112-4fde-a056-2aa9e868e863">

This will also add a new "application/lottie+zip" option when you're in the media upload / select file dialog, as seen in this screenshot:
<img width="232" alt="example-media-select-lottie-dotlottie-filter-mime-issue" src="https://github.com/LottieFiles/lottie-player/assets/13838392/902cf0ef-ccab-4ef1-ab00-4141912b0e90">

Unfortunately, I don't know of a way to change the upload filter to show a more user-friendly name like "Lottie (dotLottie)" without breaking the ability to upload .lottie files, due to what seems like a limitation in how WordPress handles custom MIME types or maybe just my own failure of imagination. Maybe someone else offer an improved solution that keeps the filter names consistent and user friendly, while also ensuring that the resulting code will still enable people to upload .lottie files.

**Comment by aliharis** [comment]
Hey everyone,

Haris here from LottieFiles. Over the past year, we've had ongoing discussions with some of our users about establishing a standard MIME type for dotLottie files. After careful consideration, we've decided to go with `application/zip+dotlottie`.

We've also updated the official documentation to reflect this change. You can check it out here: https://dotlottie.io/structure/#dotlottie-structure
